File: sample_objc/ExelbidSample/Contollers/EBVideoAdViewController.m
//
//  EBVideoAdViewController.m
//

#import "EBVideoAdViewController.h"
#import <ExelBidSDK/ExelBidSDK-Swift.h>

@interface EBVideoAdViewController ()<EBVideoDelegate>
@property (weak, nonatomic) IBOutlet UITextField *keywordsTextField;
@property (weak, nonatomic) IBOutlet UIButton *loadAdButton;
@property (weak, nonatomic) IBOutlet UIButton *showAdButton;

@property (nonatomic, strong) EBVideoManager *adManager;
@end

@implementation EBVideoAdViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.

    self.showAdButton.hidden = YES;
    self.keywordsTextField.text = self.info.ID;
    [self.loadAdButton.layer setCornerRadius:3.0f];
    [self.showAdButton.layer setCornerRadius:3.0f];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)didTapLoadButton:(id)sender
{
    [self.keywordsTextField endEditing:YES];
    
    self.spinner.hidden = NO;
    self.showAdButton.hidden = YES;
    self.loadAdButton.enabled = NO;
    self.failLabel.hidden = YES;
    
    self.adManager = [[EBVideoManager alloc] initWithIdentifier:self.keywordsTextField.text];
    [self.adManager testing:YES];
    [self.adManager yob:@"1976"];
    [self.adManager gender:@"M"];
    
    [self.adManager startWithCompletionHandler:^(EBVideoAdRequest *request, NSError *error) {
        if (error) {
            NSLog(@"Failed to load video ad with error: %@", [error localizedDescription]);
            return;
        }
        
        self.loadAdButton.enabled = YES;
        self.showAdButton.hidden = NO;
    }];
}

- (IBAction)didTapShowButton:(id)sender
{
    [self.adManager presentAdWithController:self delegate:self];
    self.showAdButton.hidden = YES;
}

#pragma mark - <EBVideoDelegate>

- (void)videoAdDidLoadWithAdUnitID:(NSString * _Nonnull)adUnitID
{
    self.spinner.hidden = YES;
    self.showAdButton.hidden = NO;
    self.loadAdButton.enabled = YES;
}

- (void)videoAdDidFailToLoadWithAdUnitID:(NSString * _Nonnull)adUnitID error:(NSError * _Nonnull)error
{
    self.failLabel.hidden = NO;
    self.loadAdButton.enabled = YES;
    self.spinner.hidden = YES;
    
}

@end

File: sample/ExelbidSample/Controller/EBVideoAdViewController.swift
//
//  EBVideoAdViewController.swift
//  com.zionbi.ExelbidSample-Swift
//
//  Created by HeroK on 2021/02/23.
//

import UIKit
import ExelBidSDK

class EBVideoAdViewController: UIViewController, EBVideoDelegate {
    @IBOutlet var keywordsTextField: UITextField!
    @IBOutlet var isTest: UIButton!
    @IBOutlet var loadAdButton: UIButton!
    @IBOutlet var showAdButton: UIButton!
  
    var info: EBAdInfoModel?
    var videoManager: EBVideoManager?
}

extension EBVideoAdViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        keywordsTextField.text = self.info?.ID
    }
    
    @IBAction func loadAdClicked(_ sender: UIButton) {
        guard let identifier = keywordsTextField.text else {
            return
        }
        
        keywordsTextField.endEditing(true)
        
        loadAdButton.isEnabled = false
        
        self.videoManager = EBVideoManager(identifier: identifier)
        
        if let videoManager = self.videoManager {
            // 광고의 효율을 높이기 위해 옵션 설정
            videoManager.yob("1987")
            videoManager.gender("M")
            
            // 테스트 광고 설정 (true - 테스트 광고가 응답)
            videoManager.testing(isTest.isSelected)

            videoManager.startWithCompletionHandler { (request, error) in
                self.loadAdButton.isEnabled = true
                
                if let error = error  {
                    print(">>> \(error.localizedDescription)")

                    "Fail Load Ad".alert(self)
                }else{
                    self.showAdButton.isEnabled = true
                    
                    "Load Ad".alert(self)
                }
            }
        }
    }
    
    @IBAction func showAdButton(_ sender: UIButton) {
        showAdButton.isEnabled = false
        
        if let videoManager = videoManager {
            videoManager.presentAd(controller: self, delegate: self)
        } else {
            "Error Video".alert(self)
        }
    }
    
    @IBAction func toggleTestButton(_ sender: UIButton) {
        sender.isSelected.toggle()
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        keywordsTextField.resignFirstResponder()
    }
}
